Monthly report 2024-04 (#5)
cardano-scaling/website: monthly/2024-04-monthly.md

---
title: April 2024
slug: 2024-04
authors: [ch1bo, quantumplation, cleanerm5, jpraynaud, ffakenz]
tags: [monthly]
---

This is a monthly report of progress on 🐲 Hydra and 🛡 Mithril projects since March 2024. This document serves as a preparation for and a written summary of the monthly stakeholder review meeting, which is announced on our Discord channels and held on Google Meet. This month, the meeting was held on 2024-04-30, using these [slides][slides] and [you can see the recording here][recording].

## Logo design work

> Throughout March and April we were looking into refreshing the logos for both Hydra and Mithril, but also create a dedicated one for common workstream of scaling cardano. Having a logo for the latter in particular is important as we hope to unite and involve multiple projects trying to improve scalability of Cardano.

Reviewer: ```suggestion
During the months of March and April, we focused on refreshing the logos of Hydra and Mithril. We also created a separate logo for the shared workstream aimed at scaling Cardano. It's crucial to have a logo for this workstream as it will help bring various projects together to enhance the scalability of Cardano.
```

![](img/2024-04-logo-drafts.png)<small><center>Latest logo drafts for Hydra, Mithril and Cardano Scaling work group</center></small>

> Above you see the latest drafts for the logos created by our designer Alexander Wende. What do you think? 🔈 Let us know through any of our communication channels like Discord or Twitter. 

Reviewer: ```suggestion
Above, you can see the latest logo drafts created by our designer, Alexander Wende. Let us know your thoughts through any of our communication channels like Discord or Twitter. 
```

## Mithril

[Issues and pull requests closed in
April](https://github.com/input-output-hk/mithril/issues?q=is%3Aclosed+sort%3Aupdated-desc+closed%3A2024-03-31..2024-04-30)

> As a part of our strategy to increase the participation in the Mithril protocol we have been reaching out to SPO's who don't yet run a Mithril signer, providing them with the support they need to set up and run their signer. We appreciate the SPO's such as Cardano Foundation who took the step in joining the protocol. As of epoch 481, the number of stake pools registered in the Mithril protocol as signers has reached 258 representing ~5 billion ADA which is ~22% of the Cardano's active stake.

Reviewer: ```suggestion
As a part of our strategy to increase the participation in the Mithril protocol we have been reaching out to SPO's who don't yet run a Mithril signer, providing them with the support they need to set up and run their signer. We appreciate the SPO's such as Cardano Foundation who took the step in joining the protocol. As of epoch 481, the number of stake pools registered in the Mithril protocol as signers has reached 258. This represents approximately 5 billion ADA, which is approximately 22% of Cardano's active stake.
```

This enhancement was initiated following a discussion led by the MLabs team, as documented [here](https://github.com/input-output-hk/hydra/discussions/1337). The need arose to unblock a [sub-validator](https://github.com/mlabs-haskell/hydra-auction-onchain/blob/95fae061a4a50a635007a7e228443080f8141570/src/HydraAuctionOnchain/Validators/StandingBid.hs#L139-L157) utilized in the hydra-auction for checking spending conditions to move a standing bid to L2. The discussion was promptly converted into an [issue](https://github.com/input-output-hk/hydra/issues/1350) and subsequently added to our roadmap after internal grooming by the Hydra team.

While this refactor represents a breaking change, the hydra-node can still be utilized as before if the provided UTxO is at a public key address. However, to spend from a script UTxO and unlock more complex use-cases, users must provide an additional unsigned transaction that accurately specifies required data, such as redeemers and validity ranges.

## Cardano Buidler Fest

> This month the who-is-who of Cardano builders got lured to Toulouse, France for a two day tech-only conference. It was a great occassion to meet fellow contributors and building bridges (literally) to new projects and their creators.

Reviewer: ```suggestion
This month the who-is-who of Cardano builders got lured to Toulouse, France for a two-day tech-only conference. It was a great occasion to meet fellow contributors and build bridges (literally) to new projects and their creators.
```

Two of many notable presentations and how they relate to the Hydra and Mithril projects are detailed as contributed by Pi (Sundae Labs) and Sebastian (Cardano Foundation).

### IBC bridge

> Cardano Foundation has started to work on an implementation of the Inter Blockchain Communication protocol ([IBC](https://github.com/cosmos/ibc)) to allow builders to connect other blockchains to the Cardano network that also implement IBC like e.g. [Cosmos SDK](https://docs.cosmos.network/) or [Substrate](https://docs.substrate.io/) based blockchains. Whereas IBC has many similarities with other communication protocols like e.g. TCP and is based on the pretty straight forward delivery of data packets from one blockchain to the other by so called Relayers the general security model is based on the assumption that the consensus of both blockchains to be connected can be trusted. The latter resolves in the challenge that so called Light Clients need to be implemented that allow one network to check the validity of the consensus of the other, to verify if a specific operation like e.g. the locking of certain assets has been done as communicated via IBC. This typically implies that a blockchain network is able to generate so called state proofs that can be presented to the Light Client which then is able to perform the validation.

Reviewer: ```suggestion
The Cardano Foundation is currently working on implementing the Inter Blockchain Communication protocol, IBC. This implementation will allow builders to connect other blockchains to the Cardano network that also implement IBC protocols like [Cosmos SDK](https://docs.cosmos.network/) or [Substrate](https://docs.substrate.io/) based blockchains.
The IBC protocol has many similarities with other communication protocols like TCP and is based on the straightforward delivery of data packets from one blockchain to another using Relayers. The general security model is based on the assumption that the consensus of both blockchains to be connected can be trusted. The latter resolves the challenge that so-called Light Clients need to be implemented that allows one network to check the validity of the consensus of the other, to verify if a specific operation, for example, the locking of certain assets has been done as communicated via IBC. This typically implies that a blockchain network is able to generate so-called state proofs that can be presented to the Light Client which then is able to perform the validation.
```

Author: How important are these suggested changes? This has been a contributed text and I would like to keep modifications of it minimal.

Reviewer: Hi @ch1bo Sorry for my late reply, it was a public holiday yesterday.
You can leave the original text unchanged if there is nothing too serious flagged.

> Unfortunately, the Cardano blockchain does not generate such state proofs as the underlying architecture and Ouroboros consensus do not rely on such a primitive. This leads to the the challenge that implementing a Light Client purely based on available data from the consensus (e.g. block data) has several shortcomings with regards to security or if traded for it liveness of the client causing significant latency in the message delivery between both networks (as the Light Client would have to wait until a sequence of blocks is presented for validation), which is not a feasible option for a real world production deployment.

Reviewer: ```suggestion
Unfortunately, the Cardano blockchain doesn't produce state proofs because its underlying architecture and Ouroboros consensus doesn't rely on it. This creates a challenge because implementing a Light Client based only on available data from the consensus (such as block data) has some limitations in terms of security. Moreover, if this approach is used for the liveness of the client, it can cause significant latency in message delivery between networks. This waiting period occurs because the Light Client needs to wait until a sequence of blocks is presented for validation. Therefore, this approach is not a feasible option for a real-world production deployment.
```

Author: How important are these suggested changes? This has been a contributed text and I would like to keep modifications of it minimal.

Reviewer: Hi @ch1bo Sorry for my late reply, it was a public holiday yesterday.
You can leave the original text unchanged if there is nothing too serious flagged.
